### DISCUSSION TITLE
App for sending data over uart (Roadrunner)

### DISCUSSION THREAD

**Discussion by Bjarne33** [bitcraze/discussions#82] [Q&A]
Hello,

I want to write an application that sends the positioning data from the roadrunner to my arduino by using uart2. I was able to print the data in the console of the cfclient. But whenever I initialize and test the uart connection, I can't connect to the cfclient anymore.
This is the current code of the main function. Why can't I connect to the cfclient, when I test the uart initialization? 

![app_code](https://user-images.githubusercontent.com/85678852/172828075-a9005d70-bbf8-4592-aecb-e675cf1dfce5.png)
 
Thank you for your help!
Best regards,
Bjarne

**Comment by Bjarne33** [comment]
update:
seems like the system wasn't ready. I added systemWaitStar() and now it works.

**Reply by krichardsson** [reply]
Great that is worked out!

`systemWaitStar()` is blocking until all sub systems have started and it is a good practice to use it in apps.
